cloudinit/util.py

```diff
@@ -1595,25 +1597,31 @@ def load_file(fname, read_cb=None, quiet=False, decode=True):
             raise
     contents = ofh.getvalue()
     LOG.debug("Read %s bytes from %s", len(contents), fname)
-    if decode:
-        return decode_binary(contents)
-    else:
-        return contents
+    return contents
+
+
+def load_text_file(
+    fname: Union[str, os.PathLike],
+    *,
+    read_cb: Optional[Callable[[int], None]] = None,
+    quiet: bool = False,
+) -> str:
+    return decode_binary(load_binary_file(fname, read_cb=read_cb, quiet=quiet))
```

Comment:
This means that every text file that we load is first read in binary mode then we manually decode it to text after it is in our buffer. Is there a reason we don't just `open(<file>, "r")` which will do that for us?

Reply:
I don't see a reason no...but I didn't want to further complicate this PR by changing any behavior. This is what was done before, so I'd rather not change more behavior on an enormous PR.

Reply:
> but I didn't want to further complicate this PR by changing any behavior

Okay, that's fair.